### PR TITLE
CXX-2988 Use Bash and shebangs whenever able

### PR DESCRIPTION
**Pull request by eramongodb**
Resolves CXX-2988. Verified by [this patch](https://spruce.mongodb.com/version/66a16c551f04b80007494da4).

Companion PR to https://github.com/mongodb/mongo-c-driver/pull/1680.

Additionally removes some stray env vars and a long-broken `sed` command in the `uninstall_check` tasks due to a combination of CXX-2753 (directory restructure) and CXX-2809 (removal of legacy CMake packages).

The `etc/generate_uninstall.sh` script is excluded from the shebang update to `bash`. It remains an `sh` script due to incompatibilities with how Bash escapes a sequence of backslashes `\\\\` in strings. This results in incorrect contents of the generated `uninstall.sh` file. The following diffs are examples of incorrect changes produced by `sh` -> `bash`:

Missing escape backslashes:

```diff
- printf "Removing file \"include/bsoncxx/fwd.hpp\""
+ printf "Removing file "include/bsoncxx/fwd.hpp""
```

Failure to expand newlines prior to sort and iteration:

```diff
- printf "Removing directory \"share/mongo-cxx-driver\""
+ printf "Removing directory "include/bsoncxx\ninclude\n[...]"
```

Refactoring the `etc/generate_uninstall.sh` script to be Bash-compatible (or revisiting its method of generation entirely) is deferred to CXX-3073.

### CODE REVIEW

**Comment by eramongodb** [comment]
Forgot to push updated commits prior to posting PR: rebased accordingly.

**Review by kevinAlbs — APPROVED** [review]

File: generate_uninstall/CMakeLists.txt [hunk 1]
```diff
@@ -68,10 +68,32 @@ else ()
          OUTPUT_FILE
             \"${CMAKE_CURRENT_BINARY_DIR}/${UNINSTALL_PROG}\"
       )
+
+      # Ensure generated uninstall script has executable permissions.
+      if (\"${CMAKE_VERSION}\" VERSION_GREATER_EQUAL \"3.19.0\")
+         file(
```
Comment (rcsanchez97) [inline generate_uninstall/CMakeLists.txt:74]:
```suggestion
         file (
```

File: generate_uninstall/CMakeLists.txt [hunk 2]
```diff
@@ -68,10 +68,32 @@ else ()
...
+            CHMOD \"${CMAKE_CURRENT_BINARY_DIR}/${UNINSTALL_PROG}\"
+            PERMISSIONS
+                  OWNER_READ OWNER_WRITE OWNER_EXECUTE
+                  GROUP_READ GROUP_EXECUTE
+                  WORLD_READ WORLD_EXECUTE
+         )
+      else ()
+         # Workaround lack of file(CHMOD).
+         file(
```
Comment (rcsanchez97) [inline generate_uninstall/CMakeLists.txt:83]:
```suggestion
         file (
```

File: generate_uninstall/CMakeLists.txt [hunk 3]
```diff
@@ -68,10 +68,32 @@ else ()
...
+            COPY \"${CMAKE_CURRENT_BINARY_DIR}/${UNINSTALL_PROG}\"
+            DESTINATION \"${UNINSTALL_PROG}.d\"
+            FILE_PERMISSIONS
+                  OWNER_READ OWNER_WRITE OWNER_EXECUTE
+                  GROUP_READ GROUP_EXECUTE
+                  WORLD_READ WORLD_EXECUTE
+         )
+         file(RENAME \"${UNINSTALL_PROG}.d/${UNINSTALL_PROG}\" \"${UNINSTALL_PROG}\")
```
Comment (rcsanchez97) [inline generate_uninstall/CMakeLists.txt:91]:
```suggestion
         file (RENAME \"${UNINSTALL_PROG}.d/${UNINSTALL_PROG}\" \"${UNINSTALL_PROG}\")
```

**Review by rcsanchez97 — APPROVED** [review]
LGTM, with a few minor tweaks (consistent spacing before opening parenthesis)

**Pull request merged**